packages/workspace/src/generators/convert-to-buildable/convert-to-buildable.ts

```diff
@@ -0,0 +1,175 @@
+import {
+  getWorkspacePath,
+  isStandaloneProject,
+  NxJsonConfiguration,
+  readJsonFile,
+  readProjectConfiguration,
+  readWorkspaceConfiguration,
+  formatFiles,
+  Tree,
+  updateJson,
+  WorkspaceConfiguration,
+  writeJson,
+  convertNxGenerator,
+} from '@nrwl/devkit';
+
+import { Schema } from './schema';
+import { checkWorkspaceVersion } from '@nrwl/workspace/src/utils/workspace';
+import { join } from 'path';
+
+function getExecutor(type: Schema['type']): string {
+  switch (type) {
+    case 'angular':
+      return '@nrwl/angular:ng-packagr-lite';
+    case 'detox':
+    case 'js':
+    case 'web':
+      return '@nrwl/js:tsc';
+    case 'next':
+    case 'react':
+      return '@nrwl/web:rollup';
+    case 'node':
+    case 'nest':
+      return '@nrwl/node:package';
+  }
+}
+
+export async function convertToBuildable(host: Tree, schema: Schema) {
+  const workspace = readWorkspaceConfiguration(host);
+
+  checkWorkspaceVersion(workspace, host);
+
+  const configuration = readProjectConfiguration(host, schema.project);
+
```

Comment:
Can we add a check here to make sure the project is a library? We don't want to accidentally overwrite apps.